api/types/container/host_config.go

```diff
@@ -244,6 +244,16 @@ func (n PidMode) Container() string {
 	return ""
 }
 
+// DeviceRequest represents a request for devices from a device driver.
+// Used by GPU device drivers.
+type DeviceRequest struct {
+	Driver       string            // Name of device driver
+	Count        int               // Number of devices to request (-1 = All)
+	DeviceIDs    []string          // List of device IDs as recognizable by the device driver
+	Capabilities [][]string        // An OR list of AND lists of device capabilities (e.g. "gpu")
```

Comment:
We were discussing `Capabilities` as a name for this (as it could be confused for `Capabilities` on the container itself (i.e. Linux capabilities)), but I can't come up with good alternatives; perhaps `Features`, but not sure if that's a good match

Reply:
I understand it but on the other hand, it's literally a list of what the device is capable of doing, what capabilities it provides. In this case it provides "gpu" capability, as well as nvidia-specific capabilities like "compute", etc.

Reply:
The only other names I can think of is "requirements" or "constraints", but I'm unsure.

Reply:
It only matches if all of these are matched, correct? "constraints" could work, but possibly  _too_ generic? idk. Naming is really hard on this one